R/path.R

```diff
@@ -98,3 +97,44 @@ path_home <- function(...) {
 path_temp <- function(...) {
   path(tempdir(), ...)
 }
+
+#' Manipulate path extensions
+#'
+#' `path_ext()` returns the last extension (if any) for a path.
+#' `path_ext_remove()` removes the last extension and returns the rest of the
+#' path. `path_ext_set()` replaces the extension with a new extension. If there
+#' is no existing extension the new extension is appended.
+#' @template fs
+#' @param ext,value The new file extension.
+#' @export
+#' @examples
+#' path_ext("file.zip")
+#' path_ext("file.tar.gz")
+#'
+#' path_ext_remove("file.tar.gz")
+#'
+#' # Only one level of extensions is removed
+#' path_ext_set(path_ext_remove("file.tar.gz"), "zip")
+path_ext <- function(path) {
+  res <- captures(path, regexpr("(?<!^|[.])[.]([^.]+)$", path, perl = TRUE))[[1]]
+  res[is.na(res)] <- ""
```

Comment:
This made me wonder what happens if `path_tidy()` gets an `NA`. Should we warn? Error?

Reply:
I think it should just pass them along as NA.

Reply:
In that case, won't this happen:

```R
path_ext(NA)
#> [1] ""
```

Reply:
Yes, thank you for mentioning this. Now all the `path_ext()` functions are robust to NA inputs (with tests).